splinterd/src/node/running/mod.rs

```diff
@@ -90,6 +92,26 @@ impl Node {
             "foo".to_string(),
         ))
     }
+
+    pub fn stop(mut self) -> Result<RunnableNode, InternalError> {
+        let rest_api_variant = match self.rest_api_variant {
+            NodeRestApiVariant::ActixWeb1(_, _) => RestApiVariant::ActixWeb1,
+            NodeRestApiVariant::ActixWeb3(_) => RestApiVariant::ActixWeb3,
+        };
+
+        let runnable_node = NodeBuilder::new()
+            .with_node_id(self.node_id().to_string())
+            .with_rest_api_variant(rest_api_variant)
+            .with_admin_signer(self.admin_signer.to_owned())
+            .with_rest_api_port(self.rest_api_port().into())
+            .with_store_factory(self.admin_subsystem.store_factory.to_owned())
```

Comment:
This clone could be avoided by taking the store factory out of the admin subsystem (which in this function may not need a lot of Option to accomplish because it consumes self).

Reply:
See - https://doc.rust-lang.org/rust-by-example/flow_control/match/destructuring/destructure_structures.html

Reply:
Updated this method to take store factory out of admin subsystem and then do the shutdown procedure